### PR TITLE
Move OS specific code into its own folder

### PR DESCRIPTION
**Pull request by alalazo**
Move all the `llnl.*` code that is either Windows or Unix specific into separate folders. The API for both platforms is the same and the correct code is imported based on "sys.platform". 

Modifications:
- [x] Move platform specific code into separate folders
- [x] Import in `llnl.util.filesystem` and `llnl.util.symlink` the correct implementation based on the current `sys.platform`

### CODE REVIEW

**Comment by alalazo** [comment]
This is an attempt to avoid having `is_windows` appearing into multiple places within a package. If the approach seems better than the current state it can be extended to other parts in `spack.*` code. Any comment on the layout used in this PR is welcome.

File: lib/spack/llnl/util/filesystem.py [hunk 1]
```diff
@@ -84,26 +74,49 @@
 ]
 
 
-def getuid():
-    if is_windows:
-        import ctypes
+if sys.platform == "win32":
```
Comment (alalazo) [inline lib/spack/llnl/util/filesystem.py:77]: The assumption here is that we'll import different implementations of the same functions in both conditional branches.

**Comment by alalazo** [comment]
@johnwparent @BetsyMcPhail @tgamblin @scheibelp fyi let me know if this approach is worth polishing / applying to other parts of the code.

File: lib/spack/llnl/util/_windows/__init__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline lib/spack/llnl/util/_windows/__init__.py:1]:
```suggestion
# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

File: lib/spack/llnl/util/_windows/symlink.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline lib/spack/llnl/util/_windows/symlink.py:1]:
```suggestion
# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — COMMENTED** [review]
Real quick pass I noticed there are a couple of copyright end years off.

File: lib/spack/llnl/util/_windows/symlink.py (hunk 3 above)
Comment (alalazo) [reply lib/spack/llnl/util/_windows/symlink.py:1]: Great catch, I wonder where I copied those from and more importantly why CI was passing :thinking:

**Comment by scheibelp** [comment]
My general preference is that if a method can abstract Windows-specific details, then having `is_windows` checks inside of it is good and can avoid code duplication. I should add specifics here but wanted to voice that general concern.

**Comment by alalazo** [comment]
Since this was just a proof of concept, and is now outdated, close it.

**Pull request closed without merging**